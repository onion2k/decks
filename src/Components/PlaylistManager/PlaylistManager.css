
.playlist {
    position: relative;
    cursor: pointer;
    color: #666;
    background-color: #eee;
    border-bottom: 1px dotted #B0B0FB;
    padding:  10px 10px;
}

.playlist .fa {
    font-size: 14px;
    margin-right: 8px;
    color: #ccc;
}

.playlist .fa-trash {
    display: none;
}

.playlist:hover .fa-trash {
    display: inline;
    padding: 0 0 0 0;
    color: #F00;
}

.playlist:hover .fa-music, .playlist:hover .fa-list {
    display: none;
}

.playlist > span {
    float: right;
}

.playlist:hover {
    color: #3E3EAF;
}

.playlist .tracks {
    display: block;
}

.playlist .share {
    display: none;
}

.playlist:hover .tracks {
    display: none;
}

.playlist:hover .share {
    display: block;
}
