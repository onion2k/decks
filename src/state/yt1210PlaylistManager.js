import { configure, decorate, observable, action, computed } from "mobx";

configure({ enforceActions: true });

export default class yt12010PlaylistManager {

    // Playlists

    playlist = 1;
    track = 0;
    currentTrack = { videoId: "", label: "" };

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [
            { title: "", videoId: "a3Z4RWZa9WA", duration: "" },
            { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
            { title: "", videoId: "Rs38lKxmtI4", duration: "" },
            { title: "", videoId: "v_yTphvyiPU", duration: "" },
            { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
            { title: "", videoId: "FALYmqt-7TQ", duration: "" }
          ]}
    ];

    load = (id) => {
        this.playlist = id;
        // const pl = this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    // add playlist
    playlistAdd = (title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    };

    // delete playlist
    playlistDelete = (index)=>{
        this.playlists.splice(index, 1);
    };

    // add track
    playlistAddTrack = (id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    };

    // delete track
    playlistDeleteTrack = (id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    };

    get tracks() {
        return this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    updateTrackData = (data, duration) => {
        const track = this.tracks.find((track)=>track.videoId === data.video_id);
              track.title = data.title;
              track.duration = duration;
        this.currentTrack = track;
    };

    getTracks = () => {
        return this.tracks;
    };

    getCurrentTrack = () => {
        return this.tracks[this.track];
    };

    getNextTrack = () => {
        return this.tracks[this.track++];
    };

    getTrackById = (id) => {
        this.track = this.tracks.findIndex((track) => track.videoId === id);
        return this.tracks[this.track];
    };
}

decorate(yt12010PlaylistManager, {
    currentTrack: observable,
    playlists: observable,
    playlist: observable,
    track: observable,
    tracks: computed,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action,
    getCurrentTrack: action,
    getNextTrack: action
});
