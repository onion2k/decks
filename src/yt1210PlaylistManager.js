import { configure, decorate, observable, action, autorun } from "mobx";

export default class yt12010PlaylistManager {

    // Playlists

    playlist = [];

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [0,1,2,3,4,5] },
        { "id": 3, "title": "Playlist 3", tracks: [0,1,2,3,4,5] },
        { "id": 4, "title": "Playlist 4", tracks: [0,1,2,3,4,5] }
    ];

    load = action((id) => {
        this.playlist = this.playlists.find((playlist) => playlist.id === id).tracks;
        this.videoId = this.playlist[0].videoId;
        this.playing = true;
    });

    // add playlist
    playlistAdd = action((title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    });

    // delete playlist
    playlistDelete = action((index)=>{
        this.playlists.splice(index, 1);
    });

    // add track
    playlistAddTrack = action((id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    });

    // delete track
    playlistDeleteTrack = action((id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    });

    updateTrackData = action((data, duration) => {

    });

}

decorate(yt12010PlaylistManager, {
    playlist: observable,
    playlists: observable,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action
});