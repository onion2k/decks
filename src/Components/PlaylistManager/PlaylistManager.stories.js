import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import PlaylistManager from '../Components/PlaylistManager';

storiesOf('PlaylistManager', module)
  .add('Basic', () => (
    <PlaylistManager />
  ));
