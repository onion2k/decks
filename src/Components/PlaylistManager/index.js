import React, { Component } from "react";
import { inject, observer } from "mobx-react";
import { withRouter } from "react-router";

import "./PlaylistManager.css";

class PlaylistManager extends Component {
  constructor(props) {
    super(props);

    this.state = {
      newlist: "",
      playlists: props.playlists
    };

    this.handleChange = this.handleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ newlist: event.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.playlistManager.playlistAdd(this.state.newlist);
  }

  render() {
    let playlists;

    if (this.props.playlistManager.playlists.length > 0) {
      playlists = this.props.playlistManager.playlists.map(playlist => {
        return (
          <div
            key={playlist.id}
            className="playlist"
            onClick={e => {
              this.props.playlistManager.load(playlist.id);
              this.props.history.push('/');
            }}
          >
            <i
              className="fa fa-trash"
              aria-hidden="true"
              onClick={e => {
                e.stopPropagation();
                e.preventDefault();
                // this.props.deleteList(playlist.playlistId);
              }}
            />{" "}
            <i className="fa fa-list" aria-hidden="true" /> {playlist.title}{" "}
            <span className="tracks">{playlist.tracks.length || "No"} tracks</span>
            <span className="share" onClick={(e)=>{
              e.stopPropagation();
              e.preventDefault();
              console.log("Share ", playlist.id)
              this.props.playlistManager.share(playlist.id)
            }}>Share</span>
          </div>
        );
      });
    } else {
      playlists = <div className="track">No playlists found</div>;
    }

    return (
      <div className="PlaylistManager">
        <div className="PlaylistTitle">
          <span>Your playlists</span>
        </div>
        <div className="Playlist">{playlists}</div>
        <form className="NewTrack">
          <label>
            New List:
            <input type="text" value={this.state.newlist} onChange={this.handleChange} />
          </label>
          <button onClick={this.onSubmit}>Add List</button>
        </form>
      </div>
    );
  }
}

export default withRouter(inject("playlistManager")(observer(PlaylistManager)));
