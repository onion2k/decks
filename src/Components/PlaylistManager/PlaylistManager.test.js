import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom'

import PlaylistManager from ".";

import yt1210PlaylistManager from "../../state/yt1210PlaylistManager"
const playlistManagerState = new yt1210PlaylistManager();

describe("Playlist Manager", () => {
  const onClick = jest.fn();

  it("renders properly", () => {
    const playlistManager = renderer
      .create(
        <MemoryRouter>
          <PlaylistManager playlistManager={playlistManagerState} />
        </MemoryRouter>
      )
      .toJSON();
    expect(playlistManager).toMatchSnapshot();
  });
});
