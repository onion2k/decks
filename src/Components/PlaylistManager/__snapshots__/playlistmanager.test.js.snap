// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Manager renders properly 1`] = `
<div
  className="PlaylistManager"
>
  <div
    className="PlaylistTitle"
  >
    <span>
      Your playlists
    </span>
  </div>
  <div
    className="Playlist"
  >
    <div
      className="track"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-trash"
        onClick={[Function]}
      />
       
      <i
        aria-hidden="true"
        className="fa fa-list"
      />
       
      Playlist 1
       
      <span>
        6
         tracks
      </span>
    </div>
    <div
      className="track"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-trash"
        onClick={[Function]}
      />
       
      <i
        aria-hidden="true"
        className="fa fa-list"
      />
       
      Playlist 2
       
      <span>
        6
         tracks
      </span>
    </div>
  </div>
  <form
    className="NewTrack"
  >
    <label>
      New List:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button
      onClick={[Function]}
    >
      Add List
    </button>
  </form>
</div>
`;
